### ISSUE TITLE
Support .NET Core 2.1 new memory primitives like Memory<T> and Span<T>

### ISSUE THREAD

**Issue by oliverw** [StormHub/NetworkCore#78]
Due to the way NetUV's ReadableBuffer is written, reading the data into a [Pipeline](https://blogs.msdn.microsoft.com/dotnet/2018/07/09/system-io-pipelines-high-performance-io-in-net/) requires a temporary copy step. It would be preferable if this could be avoided.

The optimal solution would be of course having NetUV supporting pipelines natively. :)

**Comment by StormHub** [comment]
@oliverw yep but the catch is that those are net core specific, not *net standard 2.0*, MSFT is making things available in netstandard 2.1 :)
